src/Java.Interop/Java.Interop/JniRuntime.JniValueManager.cs

```diff
@@ -650,15 +650,19 @@ static Type GetPeerType ([DynamicallyAccessedMembers (Constructors)] Type type)
 				return GetValueMarshalerCore (type);
 			}
 
-			const string MakeGenericMethod = "Generic methods used here, should be preserved via other means.";
-
-			[UnconditionalSuppressMessage ("Trimming", "IL2060", Justification = MakeGenericMethod)]
-			[UnconditionalSuppressMessage ("AOT",      "IL3050", Justification = MakeGenericMethod)]
 			static JniValueMarshaler GetObjectArrayMarshaler (Type elementType)
 			{
+				const string makeGenericMethodMessage = "This code path is not used in Android projects.";
+
+				// FIXME: https://github.com/xamarin/java.interop/issues/1192
+				[UnconditionalSuppressMessage ("Trimming", "IL2060", Justification = makeGenericMethodMessage)]
```

Comment:
Sorry for the late review:
This is potentially a trimming problem as well.
The trimming side of this is if you do something like:
```C#
class MyType
{
     public static void GenericMethod<[DAM(PublicMethods)] T>() {
          typeof(T).GetMethod("Foo").Invoke(...);
     }
}

class TestType
{
     public static void Foo() { }
}

public void Test()
{
     // This will crash at runtime, beacuse TestType.Foo is going to be trimmed away.
     // The only warning this code generates is below
     CallT(typeof(MyType).GetMethod("GenericMethod"), typeof(TestType));
}

private void CallT(MethodInfo genericMethod, Type typearg)
{
     // IL2060 (or something like it)
     genericMethod.MakeGenericMethod(typearg).Invoke(...);
}
```

The problem is that the trimmer sees that you're instantiating a generic method, but since it doesn't know which method, it must warn in the case the generic method would have an annotation on one of its type parameters.

It is also an AOT problem, that one is if the elementType is a value type, then this might fail since there might not be code for the value type instantiation of the generic method.